github/commitstatus.go

```diff
@@ -96,14 +102,44 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 	req.Header.Set("Content-Type", "application/json")
 
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		s.log.Info(fmt.Sprintf("Http request attempt: %d out of %d", attempt+1, maxRetries))
+		resp, err := httpRequestDo(req)
+		if err != nil {
+			return err
+		}
+
+		retry, timeToSleep := checkForRetry(resp, attempt)
+		if !retry {
+			return s.checkStatus(resp, state, sha, url)
+		}
+		time.Sleep(timeToSleep)
+	}
+	return nil
```

Comment:
should I report an error here?

I don't know how this can be reached but if we reach it its an error....

Reply:
You should do something like this:
```suggestion
    s.log.Warn("Maximum number of retries reached. Giving up.")
	return s.checkStatus(resp, state, sha, url)
```

Reply:
Are you sure, we will exit here if maximum retries is reached? My understanding is that we would still exit at L114 (`return s.checkStatus(resp, state, sha, url)`)

Reply:
ah I see the discussion later in this thread

Reply:
After discussion with @odormond in private chat this issue was addressed in:
https://github.com/Pix4D/cogito/pull/118/commits/c8f7ab2d1fbd2f8ea898ee590f2500d2b7d5fe71